File: 1208-get-equal-substrings-within-budget/1208-get-equal-substrings-within-budget.py
class Solution:
    def equalSubstring(self, s: str, t: str, maxCost: int) -> int:
        i = 0
        ans = 0
        n = len(s)
        
        for j in range(n):
            maxCost -= abs(ord(s[j]) - ord(t[j]))  # Calculate the cost for the current character
            while maxCost < 0 and i <= j and i < n:  # If the cost exceeds maxCost, move the start index
                maxCost += abs(ord(s[i]) - ord(t[i]))
                i += 1
            ans = max(ans, j - i + 1)  # Calculate the maximum length of substring within the maxCost
        
        return ans

File: 1208-get-equal-substrings-within-budget/1208-get-equal-substrings-within-budget.ts
function equalSubstring(s: string, t: string, maxCost: number): number {
    let i = 0;
    let ans = 0;
    const n = s.length;

    for (let j = 0; j < n; ++j) {
        maxCost -= Math.abs(s.charCodeAt(j) - t.charCodeAt(j));
        while (maxCost < 0 && i <= j && i < n) {
            maxCost += Math.abs(s.charCodeAt(i) - t.charCodeAt(i));
            ++i;
        }
        ans = Math.max(ans, j - i + 1);
    }

    return ans;
};